packages/styled-components/src/utils/injectVar.ts

```diff
@@ -0,0 +1,28 @@
+import { Interpolation, Var } from '../types';
+
+function isObject(value: unknown) {
+  return value != null && typeof value === 'object' && Array.isArray(value) === false;
+}
+
+function isVar<Props extends object>(item: any): item is Var<Props> {
+  return isObject(item) && item.var && item.compute;
+}
+
+export default function injectVar<Props extends object>(
+  chunk: Interpolation<Props>[]
+): Interpolation<Props>[] {
+  const result: Interpolation<Props>[] = [];
+
+  for (let i = 0, len = chunk.length; i < len; i += 1) {
+    const next = chunk[i];
+    if (isVar(next)) {
+      result[i - 1] = result[i - 1] + 'var(--' + next.var + ')';
```

Comment:
There is one thing to think about - css variables accept default value, here you have no possibility to pass default value. However, I do not have ideas how to do that properly.

Reply:
We always can have a default value in another way, please have a look at this example:

```js
const Comp = styled.div<{ $$fontSize?: string }>`      
  font-size: ${{ var: 'fontSize', compute: props => props.$$fontSize || "1rm" }};
`;
```

Here `1rm` is the default value as programmatically